internal/v1/defaults.go

```diff
@@ -0,0 +1,35 @@
+package v1
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+var ErrMultipleUploadRequests = errors.New("multiple upload requests are not allowed")
+
+func imageRequestsDefaults(obj *[]ImageRequest) {
+	for i := range *obj {
+		if (*obj)[i].Architecture == "" {
+			(*obj)[i].Architecture = "x86_64"
+		}
+
+		if (*obj)[i].UploadRequest == nil {
+			(*obj)[i].UploadRequest = &UploadRequest{
+				Type: UploadTypesAwsS3,
+				Options: UploadRequest_Options{
+					union: json.RawMessage(`{"type": "aws.s3", "options": {}}`),
+				},
+			}
+		}
```

Comment:
I'm not sure if default values make much sense here. If you want to add default values, they'd need to be set per image type.

Reply:
That is a good point, but we do have the information in the request, so this could be actually inferred.